gulpfile.js

```diff
@@ -71,33 +74,36 @@ gulp.task('prepare-source', function () {
         }))
         .pipe(rename('freemius-en.po'))
         .pipe(gulp.dest(languagesFolder));
-});
+}
 
 // Push updated po resource to transifex.
-gulp.task('update-transifex', ['prepare-source'], function () {
+function update_transifex() {
+    prepare_source();
```

Comment:
The `prepare_source` is an async function. It will need some time to complete. Given that this (`prepare_source`) async function is properly implemented, we need to wait for it, before pushing the resource, else there will always be race conditions. (An old version of `freemius-en.po` will be pushed).

Since `prepare_source` is a gulp task, we can use the gulp `series` composition.

```js
function prepare_source() {
  return gulp.src(...)...
}

function push_transifex() {
  return gulp.src(languagesFolder + 'freemius-en.po').pipe(transifex.pullResource());
}

const update_transifex = series(prepare_source, push_transifex);
```

Please pardon me for any interface/syntactical mistakes I might have made.